packages/ses/src/compartment.js

```diff
@@ -227,21 +208,36 @@ export const makeCompartmentConstructor = (
     // moduleMap can be invalid in ways that cannot be detected in the
     // constructor, but these checks allow us to throw early for a better
     // developer experience.
-    for (const [specifier, aliasNamespace] of entries(moduleMap || {})) {
-      if (typeof aliasNamespace === 'string') {
+    for (const [specifier, moduleDescriptor] of entries(moduleMap || {})) {
+      if (typeof moduleDescriptor === 'string') {
         // TODO implement parent module record retrieval.
         throw TypeError(
           `Cannot map module ${q(specifier)} to ${q(
-            aliasNamespace,
+            moduleDescriptor,
           )} in parent compartment`,
         );
-      } else if (weakmapGet(moduleAliases, aliasNamespace) === undefined) {
-        // TODO create and link a synthetic module instance from the given
-        // namespace object.
-        throw ReferenceError(
-          `Cannot map module ${q(
-            specifier,
-          )} because it has no known compartment in this realm`,
+      } else if (isObject(moduleDescriptor)) {
+        if (weakmapGet(moduleAliases, moduleDescriptor) !== undefined) {
+          // No furthere validation for module namespace objects.
```

Comment:
```suggestion
          // No further validation for module namespace objects.
```
unless it is an obscure pun I'm missing ;)

Reply:
Resolving